src/config.def

```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 OPTION(std::string, cache_dir, "")
+OPTION(std::string, build_cache_dir, "")
```

Comment:
Not really a "cache" since we never really reuse data from there. It's always used as a one-off. If we recompile 50 times the same program with the same options, we'll create 50 temp folders. Maybe `compiler_temp_dir` or better name?